tests/unit/test_core.py

```diff
@@ -169,6 +170,78 @@ def test_parameterset_named_initialization():
     assert paraset_named._get_or_create_region(("Earth",)) == paraset_named._root
 
 
+def test_parameterset_get_region_str():
+    paraset = ParameterSet()
+    paraset._get_or_create_region("World|Test")
+    r = paraset._get_region("World|Test")
+    assert r.full_name == ("World", "Test")
+
+    paraset._get_or_create_region("World|Second test|Lower")
+
+    root = paraset._root
+    r2 = root.get_subregion("Second test|Lower")
+    assert r2.full_name == ("World", "Second test", "Lower")
+
+
+@pytest.mark.parametrize("view_type", ["scalar", "timeseries", "generic"])
+def test_view_str_rep(view_type):
+    paraset = ParameterSet()
+    para_name = "example"
+    if view_type == "generic":
+        v = paraset.generic(para_name)
+    elif view_type == "scalar":
+        unit = "kg"
+        v = paraset.scalar(para_name, unit)
+    elif view_type == "timeseries":
+        unit = "kg"
+        tp = create_time_points(
+            np.datetime64("2010-01-01"), np.timedelta64(365, "D"), 10, "average"
+        )
+        v = paraset.timeseries(para_name, unit, tp)
+
+    str_rep = str(v)
+
+    # TODO: decide whether to unify these
```

Comment:
Oh, good point. The "read-only" is a relic...

Reply:
Ok I leave that for you